clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy

```diff
@@ -0,0 +1,645 @@
+    }
+    numTargetPools++
+    numForwardingRules++
+    resourceMap.google_compute_target_pool[targetPoolMap.name] = targetPoolMap
+    resourceMap.google_compute_forwarding_rule[forwardingRuleMap.name] = forwardingRuleMap
+
+    return true
+  }
+
+  /*
+   * Adds health check to resource map. Returns false if health check is invalid.
+   */
+  private Boolean addHealthCheckToResourceMap(GoogleHealthCheck.View healthCheck, Map resourceMap) {
+    def healthCheckMap = [:]
+
+    if (healthCheck.name) {
+      healthCheckMap.name = healthCheck.name
+    } else {
+      //required
+      return false
+    }
+    if (healthCheck.interval) {
+      healthCheckMap.check_interval_sec = healthCheck.interval
+    }
+
+    if (healthCheck.healthyThreshold) {
+      healthCheckMap.healthy_threshold = healthCheck.healthyThreshold
+    }
+    //TODO: host
+    if (healthCheck.port) {
+      healthCheckMap.port = healthCheck.port
+    }
+    if (healthCheck.requestPath) {
+      healthCheckMap.request_path = healthCheck.requestPath
+    }
+    if (healthCheck.timeout) {
+      healthCheckMap.timeout_sec = healthCheck.timeout
+    }
+    if (healthCheck.unhealthyThreshold) {
+      healthCheckMap.unhealthy_threshold = healthCheck.unhealthyThreshold
+    }
+
+    numHealthChecks++
+    resourceMap.google_compute_http_health_check[healthCheck.name] = healthCheckMap
+    return true
+  }
+
+  /*
+   * Adds security group to resource map. Returns false if security group is invalid.
+   */
+  private Boolean addSecurityGroupToResourceMap(GoogleSecurityGroup securityGroup, Map resourceMap) {
+
+    def firewallMap = [:]
+
+    if (securityGroup.name) {
+      firewallMap.name = securityGroup.name
+    } else {
+      //required
+      return false
+    }
+    if (securityGroup.network) {
+      firewallMap.network = securityGroup.network
+    } else {
+      //required
+      return false
+    }
+    if (securityGroup.inboundRules && !securityGroup.inboundRules.isEmpty()) {
+      firewallMap.allow = []
+      securityGroup.inboundRules.each { Rule rule ->
+        def allow = [:]
+        allow.protocol = rule.protocol
+        if (rule.portRanges && !rule.portRanges.isEmpty()) {
+          allow.ports = []
+          rule.portRanges.each {Rule.PortRange range ->
+            range.startPort == range.endPort ? allow.ports << "$range.startPort" : allow.ports << "$range.startPort-$range.endPort"
+          }
+        }
+        firewallMap.allow << allow
+      }
+    } else {
+      //at least one rule is required
+      return false
+    }
+    //TODO: source_ranges
+    //TODO: source_tags
+    if (securityGroup.targetTags && !securityGroup.targetTags.isEmpty()) {
+      firewallMap.target_tags = securityGroup.targetTags
+    }
+
+    numFirewalls++
+    resourceMap.google_compute_firewall[securityGroup.name] = firewallMap
+    return true
+
+  }
+
+  /*
+   * Conversion for the instance template map, hopefully will find a better way to do this in the future.
```

Comment:
We are handling nested classes here (https://github.com/spinnaker/clouddriver/blob/master/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleInstanceProvider.groovy#L107). `GoogleLoadBalancer` has a `GoogleHealthCheck` field.